### PR TITLE
Initial device details

### PR DESCRIPTION
**Pull request by mechaffin**
# Type of change? 

- [ ] Bug fix
- [ ] New feature
- [x] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 
Displays inline in the device panel for now. Will be moved when flyout is ready.
Shows tags, methods, and properties.

_Note: This does not reshape properties to a more simplified structure to allow for the upcoming property display work that Molly will be doing._
...

**Checklist:**

- [x] All tests passed
- [x] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: src/components/pages/devices/devicesDetail/simpleListDisplay.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+export class SimpleListDisplay extends Component {
+  render() {
+    const { list } = this.props;
+    if (list) {
+      var itemList = list.map((item, idx) => {
```
Comment (spryor) [inline src/components/pages/devices/devicesDetail/simpleListDisplay.js]: Use `const` instead of `var`. Basically never use var.

File: src/components/pages/devices/devices.js [hunk 2]
```diff
@@ -21,11 +35,26 @@ export class Devices extends Component {
         {
           !!error &&
           <span className="status">
-            { t('errorFormat', { message: t(error.message, { message: error.errorMessage }) }) }
+            {t('errorFormat', { message: t(error.message, { message: error.errorMessage }) })}
           </span>
         }
-        { !error && <DeviceGrid rowData={isPending ? undefined : devices || []} /> }
```
[the remark this answers is not in the document]
Comment (mechaffin) [reply src/components/pages/devices/devices.js]: Yes.

File: src/components/pages/devices/devicesDetail/devicesDetail.scss [hunk 3]
```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+@import 'src/styles/mixins';
+@import 'src/styles/themes';
+
+.devices-flyout {
```
[the remark this answers is not in the document]
Comment (mechaffin) [reply src/components/pages/devices/devicesDetail/devicesDetail.scss]: done.

File: src/components/pages/devices/devicesDetail/deviceDetailSection.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import './devicesDetail.css';
+import { SectionHeader, SectionDesc, FormSection } from 'components/shared';
+
+export const DevicesDetailSection = (props) => (
+  <div className='devices-detail-section'>
```
[the remark this answers is not in the document]
Comment (mechaffin) [reply src/components/pages/devices/devicesDetail/deviceDetailSection.js]: Yes, that's why I asked you about these beforehand.  Though, I could have looked closer at the styles themselves.  Or we could add this type of info to a readme type file so we don't have to read a bunch of code just to learn how something is supposed to be used.

File: src/components/pages/devices/devicesDetail/devicesDetail.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (mechaffin) [reply src/components/pages/devices/devicesDetail/devicesDetail.scss]: done.

File: src/components/pages/devices/devicesDetail/devicesDetail.scss [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  padding: 20px;
+  display: flex;
+  -ms-flex-flow: column nowrap;
```
[the remark this answers is not in the document]
Comment (mechaffin) [reply src/components/pages/devices/devicesDetail/devicesDetail.scss]: I'll try it. For many of these, I was "borrowing" from what was in the browser on the current site.  So its entirely possible that SASS will take care of it.

File: src/components/pages/devices/devicesDetail/simpleNameValueDisplay.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+      var pairList = Object.entries(pairs).map(([key, value]) => {
+        return (
+          <tr key={key}>
+            <td>{key} </td>
+            <td>{value.toString()}</td>
+          </tr>
+        );
+      });
+      return (
+        <table>
```
[the remark this answers is not in the document]
Comment (mechaffin) [reply src/components/pages/devices/devicesDetail/simpleNameValueDisplay.js]: Hahahaha.  Sure thing.  I was just literally copying what's on the current site.

File: src/components/pages/devices/devicesDetail/simpleListDisplay.js (hunk 1 above)
Comment (mechaffin) [reply src/components/pages/devices/devicesDetail/simpleListDisplay.js]: ok

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetailsTemp.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import { PropertyList } from './propertyList';
+
+export class DeviceDetailsTemp extends Component {
+  render() {
+    const { t, device } = this.props;
+    return (
+      !!device &&
+      <div className="device-details-container">
+        <Section title="Rules">
+          <div>TODO: Add Rules Grid. Remove section.</div>
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceDetails/deviceDetailsTemp.js]: Remove div, the text should enough.

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetailsTemp.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        </Section>
+        <Section title={t('devices.details.telemetry.title')} description={t('devices.details.telemetry.description')}>
+          <div>TODO: Add chart when able.</div>
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceDetails/deviceDetailsTemp.js]: Remove div

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetailsTemp.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+        </Section>
+        <Section title={t('devices.details.tags.title')} description={t('devices.details.tags.description')}>
+          <PropertyList t={t} pairs={device.tags} />
+        </Section>
+        <Section title={t('devices.details.methods.title')} description={t('devices.details.methods.description')}>
+          <MethodList t={t} list={!!device.methods && device.methods.split(',')} />
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceDetails/deviceDetailsTemp.js]: {(device.methods || []).split(',')}

File: src/components/pages/devices/flyouts/deviceDetails.js [hunk 10]
```diff
@@ -9,17 +9,41 @@ import {
   FlyoutContent
 } from 'components/shared';
 
+import './deviceDetails/deviceDetails.css';
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceDetails.js]: See the comment in the `deviceDetailsTemp.js` file about the import ordering pattern.

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetails.scss [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    .section-content { padding: 0 1rem 0 1rem; }
+
+    .property-row {
+      display: flex;
+
+      .property-name {
+        padding-inline-end: 0.5rem;
+        width: 30% !important;
```
[the remark this answers is not in the document]
Comment (mechaffin) [reply src/components/pages/devices/flyouts/deviceDetails/deviceDetails.scss]: My "column" alignment wasn't working.  At some point it did work... but may have been unrelated to adding !important.  Since it seems to work now without it.

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+    ];
+  }
+
+  componentDidMount() {
+    if (!this.props.rulesLastUpdated) this.props.fetchRules();
+    this.fetchAlarms((this.props.device || {}).id)
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.fetchAlarms((nextProps.device || {}).id)
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js]: I think you should check the device Ids here. Not totally sure but I think if you click the same device row twice it will still cause the props to update here. If the device ids are the same no need to call fetchAlarms again.

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js [hunk 13]
```diff
@@ -0,0 +1,169 @@
...
+    }));
+
+  fetchAlarms = (deviceId) => {
+    this.setState({ isAlarmsPending: true });
+    TelemetryService.getAlarms({
+      limit: 5,
+      order: "desc",
+      devices: deviceId
+    })
+      .subscribe(
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js]: On thing to note is that whenever a subscribe occurs, it returns a subscription object. You need to save that object and call `.unsubscribe` on it in `componentWillUnmount()`. If a user closes the flyout before the call completes and the subscription isn't unsubscribed then the onNext or onError function will be called. Those will try to call `setState` on a component that doesn't exist anymore, causing an error.

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js [hunk 14]
```diff
@@ -0,0 +1,169 @@
...
+              <FlyoutSection>
+                <SectionHeader>{t('devices.details.tags.title')}</SectionHeader>
+                <SectionDesc>{t('devices.details.tags.description')}</SectionDesc>
+                <Grid className="section-content">
+                  <Row>
+                    <HeaderCell className="col-3">{t('devices.details.tags.keyHeader')}</HeaderCell>
+                    <HeaderCell className="col-7">{t('devices.details.tags.valueHeader')}</HeaderCell>
+                  </Row>
+                  {
+                    (Object.entries(device.tags) || []).map((item, idx) =>
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js]: `.map(([propertyName, propertyValue], idx)`

or something similar, it will be easier to read

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js [hunk 15]
```diff
@@ -0,0 +1,169 @@
...
+                  }
+                </Grid>
+              </FlyoutSection>
+
+              <FlyoutSection>
+                <SectionHeader>{t('devices.details.methods.title')}</SectionHeader>
+                <SectionDesc>{t('devices.details.methods.description')}</SectionDesc>
+                <Grid className="section-content">
+                  {
+                    ((device.methods || '').split(',') || []).map((item, idx) =>
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js]: rename item to `method`

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js [hunk 16]
```diff
@@ -0,0 +1,169 @@
...
+              <FlyoutSection>
+                <SectionHeader>{t('devices.details.properties.title')}</SectionHeader>
+                <SectionDesc>{t('devices.details.properties.description')}</SectionDesc>
+                <Grid className="section-content">
+                  <Row>
+                    <HeaderCell className="col-3">{t('devices.details.properties.keyHeader')}</HeaderCell>
+                    <HeaderCell className="col-7">{t('devices.details.properties.valueHeader')}</HeaderCell>
+                  </Row>
+                  {
+                    (Object.entries(device.properties) || []).map((item, idx) =>
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js]: same comment about destructuring `item`

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetails.scss [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+
+  .flyout-section {
+    .section-content {
+      @include rem-fallback(padding, 0px, 12px, 0px, 12px);
+
+      .basic-row {  @include rem-fallback(padding-top, 16px); }
+    }
+  }
+
+  .device-icon svg {
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceDetails/deviceDetails.scss]: Seems like this should be placed inside some kind of device details header section?

File: src/components/shared/basicGrid/basicCell.js [hunk 18]
```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft. All rights reserved.
```
Comment (spryor) [inline src/components/shared/basicGrid/basicCell.js]: I'm not a fan of the name `basic`. Seems like this is generalizing too early. I'd say just go with PropertyGrid or something similar and put in back with the deviceDetails grid folder structure.

File: src/components/shared/basicGrid/styles/basicGrid.scss [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+@import 'src/styles/mixins';
+@import 'src/styles/themes';
+
+.basic-grid-container {
+  .basic-row {
+    display: flex;
+    flex-direction: row;
+
+    .basic-cell-header {
+      font-family: $fontSelawikBold;
```
Comment (spryor) [inline src/components/shared/basicGrid/styles/basicGrid.scss]: The grid doesn't need to have styles like this if it is a generic grid. The component that uses it should set any font sizes, etc.

File: src/components/shared/cellRenderers/cellRenderer.scss [hunk 20]
```diff
@@ -16,12 +16,25 @@ $iconSize: 14px;
...
+  }
+
   .pcs-renderer-text {
     white-space: nowrap;
     overflow: hidden;
     text-overflow: ellipsis;
 
     &.uppercase { text-transform: uppercase; }
+
+    >.icon-only {
```
Comment (spryor) [inline src/components/shared/cellRenderers/cellRenderer.scss]: missing space after `>`

File: src/components/shared/cellRenderers/linkRenderer/linkRenderer.js [hunk 21]
```diff
@@ -0,0 +1,21 @@
...
+
+import { Svg } from 'components/shared/svg/svg';
+import { svgs } from 'utilities';
+
+import '../cellRenderer.css';
+
+export const LinkRenderer = ({ uri, context: { t }, svgPath }) => {
+  return (
+    <div className="pcs-renderer-cell">
+      <NavLink to={uri} className="pcs-renderer-link">
```
Comment (spryor) [inline src/components/shared/cellRenderers/linkRenderer/linkRenderer.js]: Rename `uri` to `to` to maintain a consistent `to` `path` scheme following the `react-router` naming pattern.

File: src/components/shared/cellRenderers/linkRenderer/linkRenderer.js [hunk 22]
```diff
@@ -0,0 +1,21 @@
...
+
+import React from "react";
+import { NavLink } from 'react-router-dom';
+
+import { Svg } from 'components/shared/svg/svg';
+import { svgs } from 'utilities';
+
+import '../cellRenderer.css';
+
+export const LinkRenderer = ({ uri, context: { t }, svgPath }) => {
```
Comment (spryor) [inline src/components/shared/cellRenderers/linkRenderer/linkRenderer.js]: `context` and `t` can be removed

File: src/polyfills.js [hunk 23]
```diff
@@ -75,3 +75,58 @@ if (!String.prototype.padStart) {
     }
   };
 }
+
+// From https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys
+if (!Object.keys) {
```
Comment (spryor) [inline src/polyfills.js]: Why did you add keys? Keys is already supported natively by all modern browsers.

File: src/services/telemetryService.js [hunk 24]
```diff
@@ -16,8 +16,14 @@ export class TelemetryService {
       .map(toRulesModel);
   }
 
-  /** Returns a list of alarms */
+  /** Returns a list of alarms (all statuses) */
   static getAlarms(params = {}) {
+    return HttpClient.get(`${ENDPOINT}alarms?${stringify(params)}`)
+      .map(toAlarmsModel)
```
Comment (spryor) [inline src/services/telemetryService.js]: Looking at the response objects it doesn't seem like `toAlarmsModel` can be resused here.

File: src/utilities/svgs.js [hunk 25]
```diff
@@ -37,6 +43,14 @@ export const svgs = {
     maintenance: TabMaintenanceIconPath,
     rules: TabRulesIconPath
   },
+  devices: {
+    default: DeviceDefaultIconPath,
```
Comment (spryor) [inline src/utilities/svgs.js]: Rename `default` icon to `generic` It may be that in some use cases other icons are used as defaults so `generic` is more descriptive in telling you what the icon is rather than how it is used.

File: public/locales/en/translations.json [hunk 26]
```diff
@@ -33,12 +33,43 @@
...
+        "description": "Diagnostic information from device"
+      },
+      "methods": {
+        "title": "Methods",
+        "description": "All methods available for device"
+      },
+      "properties": {
+        "title": "Properties",
+        "description": "Selected properties on device",
+        "keyHeader": "PROPERTY",
```
Comment (spryor) [inline public/locales/en/translations.json]: See below comment about caps

File: src/components/shared/basicGrid/styles/basicGrid.scss [hunk 27]
```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft. All rights reserved.
```
Comment (spryor) [inline src/components/shared/basicGrid/styles/basicGrid.scss]: The style sheet should be a sibling file to the  .js file.

File: src/assets/icons/deviceChiller.svg [hunk 28]
```diff
@@ -0,0 +1,14 @@
+<!-- Copyright (c) Microsoft. All rights reserved. -->
+
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 31.441 23.892">
+  <title>Chiller </title>
```
[the remark this answers is not in the document]
Comment (mechaffin) [reply src/assets/icons/deviceChiller.svg]: Sure, I just pulled it from the old project. I'll remove the icky "Asset x" titles from the other device icons as well.

File: src/assets/icons/deviceIcon.svg [hunk 29]
```diff
@@ -0,0 +1,16 @@
+<!-- Copyright (c) Microsoft. All rights reserved. -->
+
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="-20836 -16624 89.406 72">
+  <defs>
```
[the remark this answers is not in the document]
Comment (mechaffin) [reply src/assets/icons/deviceIcon.svg]: Yup, again... this is from the old project. I didn't invent it. Didn't even look at the contents of the files since they were fromthe old... but I'm happy to fix it. :)

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js [hunk 30]
```diff
@@ -0,0 +1,169 @@
...
+      name: (rules[alarm.ruleId] || {}).name
+    }));
+
+  fetchAlarms = (deviceId) => {
+    this.setState({ isAlarmsPending: true });
+    TelemetryService.getAlarms({
+      limit: 5,
+      order: "desc",
+      devices: deviceId
+    })
```
[the remark this answers is not in the document]
Comment (mechaffin) [reply src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js]: We'll have to figure out why/how my settings are different from yours.  Since this is how my alt-shft-F formats it.

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetails.scss [hunk 31]
```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+@import 'src/styles/mixins';
+@import 'src/styles/themes';
+
+.device-details-container {
+  @include rem-font-size(14px);
+  @include rem-fallback(line-height, 17px);
```
[the remark this answers is not in the document]
Comment (mechaffin) [reply src/components/pages/devices/flyouts/deviceDetails/deviceDetails.scss]: Trying to follow the redlines.  But perhaps, I don't know enough about the right way to do this.

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetails.scss (hunk 17 above)
Comment (mechaffin) [reply src/components/pages/devices/flyouts/deviceDetails/deviceDetails.scss]: Its possible that was there before I added the header section class.

File: src/components/shared/basicGrid/basicCell.js (hunk 18 above)
Comment (mechaffin) [reply src/components/shared/basicGrid/basicCell.js]: I don't like the name either.  Since I'm planning to use it in the other flyouts, it doesn't feel too early.  But if you don't want it general now, I'll move it back under deviceDetails for now.

File: src/components/shared/basicGrid/styles/basicGrid.scss (hunk 27 above)
Comment (mechaffin) [reply src/components/shared/basicGrid/styles/basicGrid.scss]: It wasn't working that way.   So I followed the pattern used in the other shared components.  Though, since you want me to rearrange all these files, I'll try it again.

File: src/polyfills.js (hunk 23 above)
Comment (mechaffin) [reply src/polyfills.js]: Because I wasn't sure.... and the place that I got entries from had it as well.

File: src/services/telemetryService.js (hunk 24 above)
Comment (mechaffin) [reply src/services/telemetryService.js]: The only thing missing is the count.  But I can make a new one

File: src/components/shared/flyout/flyoutSection.js [hunk 32]
```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft. All rights reserved.
```
Comment (spryor) [inline src/components/shared/flyout/flyoutSection.js]: This file should not be in the `shared/flyout` folder. The only files here are those that pertain to the flyout itself. Not items that can be passed into the children for the flyout component.

File: src/components/shared/flyout/flyoutSection.js [hunk 33]
```diff
@@ -0,0 +1,60 @@
...
+
+  render() {
+    const { className, title, description, children } = this.props;
+    const displayContent = this.state.showContent ? {} : { display: 'none' };
+    return (
+      <div className={joinClasses('flyout-section', className)}>
+        <SectionHeader>
+          <div className="section-title" onClick={this.toggleContent}>{title}</div>
+          {
+            this.state.showContent
```
Comment (spryor) [inline src/components/shared/flyout/flyoutSection.js]: Move the `this.state.showContent` ternary operator into the className

File: src/components/shared/cellRenderers/cellRenderer.scss [hunk 34]
```diff
@@ -16,12 +16,25 @@ $iconSize: 14px;
...
+  }
+
   .pcs-renderer-text {
     white-space: nowrap;
     overflow: hidden;
     text-overflow: ellipsis;
 
     &.uppercase { text-transform: uppercase; }
+
+    > .icon-only {
```
Comment (spryor) [inline src/components/shared/cellRenderers/cellRenderer.scss]: oneline

File: src/components/pages/devices/flyouts/deviceDetails/propertyGrid/propertyGrid.scss [hunk 35]
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+@import 'src/styles/mixins';
+@import 'src/styles/themes';
+
+.grid-container {
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceDetails/propertyGrid/propertyGrid.scss]: `grid-container` needs to be renamed `property-grid-container`. If this is checked in it will result in a name collision with the grid used in the dashboard

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetails.scss [hunk 36]
```diff
@@ -0,0 +1,52 @@
...
+    .device-simulated { @include rem-fallback(font-size, 10px); }
+    .device-connected { @include rem-fallback(font-size, 14px); }
+
+    .device-icon svg {
+      @include rem-fallback(height, 72px);
+      @include rem-fallback(margin-right, 24px);
+    }
+  }
+
+  @include themify($themes) {
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceDetails/deviceDetails.scss]: I aim to have only one call to themify per sass file. It is easier to modify if all the theming vars are in one place and one location (the end of the file)

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js [hunk 37]
```diff
@@ -0,0 +1,173 @@
...
+    ];
+  }
+
+  componentDidMount() {
+    if (!this.props.rulesLastUpdated) this.props.fetchRules();
+    this.fetchAlarms((this.props.device || {}).id);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if ((this.props.device || {}).id === nextProps.device.id) {
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js]: This doesn't look right. It seems to be saying to fetch the alarms when the old and new device ids are the same. It should update when the ids are different. Right?

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js [hunk 38]
```diff
@@ -0,0 +1,173 @@
...
+
+  applyRuleNames = (alarms, rules) =>
+    alarms.map(alarm => ({
+      ...alarm,
+      name: (rules[alarm.ruleId] || {}).name
+    }));
+
+  fetchAlarms = (deviceId) => {
+    this.setState({ isAlarmsPending: true });
+    const getAlarmsObsv =
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js]: No reason to save this in a variable.

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js [hunk 39]
```diff
@@ -0,0 +1,173 @@
...
+      TelemetryService.getAlarms({
+        limit: 5,
+        order: "desc",
+        devices: deviceId
+      });
+
+    this.getAlarmsObsvStream = getAlarmsObsv
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js]: Rename `getAlarmsObsvStream` to `subscription`

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js [hunk 40]
```diff
@@ -0,0 +1,173 @@
...
+      .subscribe(
+      alarms => this.setState({ alarms, isAlarmsPending: false, alarmsError: undefined }),
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js]: indentation

File: src/components/shared/flyout/flyoutSection.js [hunk 41]
```diff
@@ -0,0 +1,60 @@
...
+    };
+
+    this.toggleContent = this.toggleContent.bind(this);
+  }
+
+  toggleContent() {
+    this.setState({ showContent: !this.state.showContent });
+  }
+
+  render() {
```
Comment (spryor) [inline src/components/shared/flyout/flyoutSection.js]: It would be better to implement a provider pattern here so the local accordion state is captured in the component context. Then we can keep the nice declarative approach of making sections.

File: src/utilities/svgs.js [hunk 42]
```diff
@@ -37,7 +44,16 @@ export const svgs = {
...
+  devices: {
+    generic: DeviceGenericIconPath,
+    chiller: DeviceChillerIconPath,
+    elevator: DeviceElevatorIconPath,
+    engine: DeviceEngineIconPath,
+    prototyping: DevicePrototypingIconPath,
+    truck: DeviceTruckIconPath
+  },
   carat: CaratIconPath,
+  chevron: ChevronIconPath,
```
[the remark this answers is not in the document]
Comment (mechaffin) [reply src/utilities/svgs.js]: In the old code, there are no up and down chevron.  They use chevron (not chevronRight) and rotate it.  I did the same thing.

File: src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js (hunk 40 above)
Comment (mechaffin) [reply src/components/pages/devices/flyouts/deviceDetails/deviceDetails.js]: Sure, but let's check your settings against mine... because I can change the indentation manually... but the automatic is just going to change it back.

File: src/components/shared/flyout/flyoutSection.js (hunk 41 above)
Comment (mechaffin) [reply src/components/shared/flyout/flyoutSection.js]: I am not sure what you mean in this context.
